examples/MagicWeather/app/src/main/java/com/revenuecat/sample/ui/paywall/PaywallAdapter.kt

```diff
@@ -35,19 +38,90 @@ class PaywallAdapter(
         val view = LayoutInflater.from(viewGroup.context)
             .inflate(R.layout.paywall_item, viewGroup, false)
 
-        return PackageViewHolder(view, didChoosePackage)
+        return PackageViewHolder(view, didChoosePaywallItem)
     }
 
     override fun onBindViewHolder(viewHolder: PackageViewHolder, position: Int) {
-        offering?.availablePackages?.get(position)?.let {
+        theItems[position].let {
             val item = it
             viewHolder.item = item
 
-            viewHolder.view.findViewById<TextView>(R.id.paywall_item_title).text = item.product.title
-            viewHolder.view.findViewById<TextView>(R.id.paywall_item_description).text = item.product.description
-            viewHolder.view.findViewById<TextView>(R.id.paywall_item_price).text = item.product.originalPrice
+            when (it) {
+                is PaywallItem.Title -> {
+                    viewHolder.view.findViewById<TextView>(R.id.paywall_item_options_title).visibility = View.VISIBLE
+                    viewHolder.view.findViewById<TextView>(R.id.paywall_item_options_title).text = it.title
+                    viewHolder.view.findViewById<ConstraintLayout>(R.id.purchasableLayout).visibility = View.GONE
+                }
+                is PaywallItem.Product -> {
```

Comment:
there's a way to create a different ViewHolder for each of the types, so you don't have to do this hiding/showing of certain layout fields

Reply:
though, on second thought, these layouts are basically the same, so probably not worth it. we can use databinding to show/hide and that'll clean it up